### DISCUSSION TITLE
有没有推荐配合使用的UI组件？

### DISCUSSION THREAD

**Discussion by FalconWu2017** [ofajs/discussion#17] [General]
比如EUI？

**Comment by kirakiray** [comment]
有，不过还在内测中，没有文档，需要参照 case 来使用。

https://github.com/ofajs/Punch-UI

**Reply by FalconWu2017** [reply]
ofa和bootstrap配合怎么样？毕竟bootstrap对代码侵入性很低

**Reply by kirakiray** [reply]
> ofa和bootstrap配合怎么样？毕竟bootstrap对代码侵入性很低

配合起来不错，记得组件或页面模块要使用bs样式时，`<link />` 引用 bootstrap 的库就行。

**Reply by FalconWu2017** [reply]
需要每个组件都引入bs样式和js？

**Reply by kirakiray** [reply]
> 需要每个组件都引入bs样式和js？

是的。

**Comment by kirakiray** [comment]
~~也推荐使用 [shadcn](https://ui.shadcn.com/examples/dashboard) 。它也是用 web component 开发，和 ofa.js 无缝兼容。~~

~~https://ui.shadcn.com/examples/dashboard~~

**Reply by kirakiray** [reply]
很可惜，今天再看代码，已经没有发现直接 esmodule引入使用的方法了，不再建议使用 shadcn。

**Comment by NoAntiMage** [comment]
ofajs集成bootstrap.js的问题。
我是用bootstrap-5.3.0，bs的js定义了$符号，不知道是不是有冲突。

```
const getjQuery = () => {
    if (window.jQuery && !document.body.hasAttribute('data-bs-no-jquery')) {
      return window.jQuery;
    }
    return null;
  };
const $ = getjQuery();

```

我创建了组件，实现折叠效果collapse，折叠无效。ofajs在index.html主页引入，bs在组件内引入。调用了$.noConflict()，应该对bs不生效

```
<template page>
    <link rel="stylesheet" href="/vendor/bootstrap-5.3.0-alpha1-dist/css/bootstrap.min.css">
    <link rel="stylesheet" href="/vendor/bootstrap-icons-1.11.3/font/bootstrap-icons.min.css">
    
  <div class="container">
    <button class="btn btn-primary" 
      data-bs-target="#collapseTarget" 
      data-bs-toggle="collapse">
      Bootstrap collapse
    </button>
    <div class="collapse py-2" id="collapseTarget">
      This is the toggle-able content!
    </div>
  </div>

  <script src="/vendor/bootstrap-5.3.0-alpha1-dist/js/bootstrap.bundle.min.js"></script>
  <script>
      $.noConflict();
  </script>
</template>
```

**Reply by kirakiray** [reply]
web component 影子节点内，没办法使用script标签，所以 `<script src="/vendor/bootstrap-5.3.0-alpha1-dist/js/bootstrap.bundle.min.js"></script>` 没有生效。当前只能使用 bootstrap css

**Comment by kirakiray** [comment]
现在开始，我推荐使用 https://soberjs.com/
